src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -1030,646 +1301,853 @@ public enum SupremeVictors implements LogicCardInfo {
-              damage 0
+              damage 50
             }
           }
-
         };
       case LOPUNNY_63:
-        return evolution (this, from:"Buneary", hp:HP080, type:COLORLESS, retreatCost:1) {
+        return evolution (this, from:"Buneary", hp:HP080, type:C, retreatCost:1) {
           weakness F, PLUS20
           move "Ice Beam", {
             text "20 damage. Flip a coin. If heads, the Defending Pokémon is now Paralyzed."
             energyCost C, C
             attackRequirement {}
             onAttack {
               damage 20
-              flip { apply PARALYZED }
+              flip {
+                applyAfterDamage PARALYZED
+              }
             }
           }
           move "Rear Kick", {
             text "40 damage. "
             energyCost C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 40
             }
           }
-
         };
       case LOUDRED_64:
-        return evolution (this, from:"Whismur", hp:HP080, type:COLORLESS, retreatCost:1) {
+        return evolution (this, from:"Whismur", hp:HP080, type:C, retreatCost:2) {
           weakness F, PLUS20
           move "Smash Kick", {
             text "30 damage. "
             energyCost C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
             }
           }
           move "Stomp Off", {
-            text "50 damage. Discard the top card from your opponent’s deck."
+            text "50 damage. Discard the top card from your opponent's deck."
             energyCost C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 50
+              afterDamage {
+                if(opp.deck) {
+                  discard opp.deck.first()
+                }
+              }
             }
           }
-
         };
       case MAGMAR_65:
-        return basic (this, hp:HP070, type:FIRE, retreatCost:1) {
+        return basic (this, hp:HP070, type:R, retreatCost:1) {
           weakness W, PLUS20
           move "Smokescreen", {
-            text "If the Defending Pokémon tries to attack during your opponent’s next turn, your opponent flips a coin. If tails, this attack does nothing."
+            text "If the Defending Pokémon tries to attack during your opponent's next turn, your opponent flips a coin. If tails, that attack does nothing."
             energyCost C
             attackRequirement {}
             onAttack {
               sandAttack(thisMove)
             }
           }
           move "Fireworks", {
-            text "30 damage. Energy attached to Magmar."
-            energyCost R, C, R
+            text "30 damage. Flip a coin. If tails, discard a [R] Energy attached to Magmar."
+            energyCost R, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              flip 1, {}, {
```

Comment:
I agree that the empty closures aren't aesthetically pleasing, but personally think that having no effect on heads and an effect on tails is more clear than using flip for heads and flipTails for tails.

Reply:
each closure whether empty (i.e. `{}`) or not takes up memory and disk space as each closure is compiled as a java class and they are all loaded to memory in runtime. that's the reason the server takes too much time to boot, it's loading hundreds of thousands of classes into memory. the solution is rather radical, rather than putting set implementation files in the same folder with java sources, they could be split into groovy script files for each card then loaded to a GroovyShell created specifically for each game, that would evaluate the card scripts per game, with a new classloader per shell, then throw it away at the end of game.